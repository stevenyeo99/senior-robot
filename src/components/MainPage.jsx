import { Component } from 'react';

import SearchBox from "./SearchBox";
import CardList from "./CardList";
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';

import './MainPage.css';

class MainPage extends Component {

    componentDidMount() {
        this.props.setRobots();
    }

    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchValue.toLowerCase());
        });
    }

    render() {
        return (this.props.isPending) ?
        <h1>Loading</h1>
        :
        (
            <div className='tc'>
                <Header />
                <SearchBox searchValue={this.props.searchValue} searchHandler={this.props.setSearchValue} />
                <Scroll>
                    <ErrorBoundry>
                        <CardList items={this.filteredRobots()} />
                    </ErrorBoundry>
                </Scroll>
            </div>
        );
    }
};

export default MainPage;