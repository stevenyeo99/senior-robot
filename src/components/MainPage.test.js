import { shallow, mount } from 'enzyme';
import MainPage from './MainPage';

let mainWrapper;
beforeEach(() => {
    const mockProps = {
        searchValue: '',
        setSearchValue: jest.fn(),
        robots: [],
        isPending: false,
        setRobots: jest.fn()
    };

    mainWrapper = shallow(<MainPage {...mockProps} />);
});

it('test main wrapper is succesfull rendering or not.', () => {
    expect(mainWrapper).toMatchSnapshot();
});

it('test filtered robots', () => {
    const mockProps2 = {
        searchValue: 'a',
        setSearchValue: jest.fn(),
        robots: [
            {
                id: 1,
                name: 'steven',
                email: 'yeomandeveloper@gmail.com'
            }
        ],
        isPending: false,
        setRobots: jest.fn()
    };

    const mainWrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(mainWrapper2.instance().filteredRobots()).toEqual([]);
});

it('test pending result main wrapper', () => {
    const mockProps3 = {
        searchValue: 'a',
        setSearchValue: jest.fn(),
        robots: [],
        isPending: true,
        setRobots: jest.fn()
    };

    const mainWrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(mainWrapper3.instance().props.isPending).toBe(true);
});