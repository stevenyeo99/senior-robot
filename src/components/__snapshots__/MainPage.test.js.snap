// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test filtered robots 1`] = `
<MainPage
  isPending={false}
  robots={
    Array [
      Object {
        "email": "yeomandeveloper@gmail.com",
        "id": 1,
        "name": "steven",
      },
    ]
  }
  searchValue="a"
  setRobots={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  setSearchValue={[MockFunction]}
>
  <div
    className="tc"
  >
    <Header>
      <div>
        <h1
          className="f1"
        >
          RoboFriends
        </h1>
        <CounterButton
          color="red"
        >
          <button
            color="red"
            id="button"
            onClick={[Function]}
          >
            Count: 
            0
          </button>
        </CounterButton>
      </div>
    </Header>
    <SearchBox
      searchHandler={[MockFunction]}
      searchValue="a"
    >
      <div
        className="pa2"
      >
        <input
          className="pa3 ba b--green bg-lightest-blue"
          onChange={[MockFunction]}
          placeholder="search robots"
          type="search"
          value="a"
        />
      </div>
    </SearchBox>
    <Scroll>
      <div
        style={
          Object {
            "border": "5px solid black",
            "height": "800px",
            "overflowY": "scroll",
          }
        }
      >
        <ErrorBoundry>
          <div>
            <CardList
              items={Array []}
            >
              <div />
            </CardList>
          </div>
        </ErrorBoundry>
      </div>
    </Scroll>
  </div>
</MainPage>
`;

exports[`test main wrapper is succesfull rendering or not. 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchHandler={[MockFunction]}
    searchValue=""
  />
  <Scroll>
    <ErrorBoundry>
      <CardList
        items={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
